### ISSUE TITLE
How to enable/make visible on Windows 11?

### ISSUE THREAD

**Issue by dacowan** [stefankueng/tools#55]
On the most recent preview flight of Windows 11 - StExBar64 is installed, but the option to show the toolbar is no longer in the explorer menu options (that I can see).  How do I enable the toolbar ribbon?
I'm sure I'm just missing something obvious, I just can't see where...

**Comment by stefankueng** [comment]
open explorer, then select the control panel in the left tree view.
The old control panel is then shown, which still has the old look including a menu. right-click on that menu bar and you'll get the menu to show StExBar. Once activated it will be visible also in all other explorer views.

**Comment by dacowan** [comment]
OK, that helped - but there's a step missing.

1. Open a Windows Explorer window
2. Select "..." in the Ribbon and select "Options"
3. Change the "View" tab, and select "Files & Folders > Always Show Menus"
4. Scroll down and find "Navigation Pane > Show All Folders"
5. Exit the dialog
6. Return to the Windows Explorer window, and locate the "Control Panel" on the left navigation pane
7. There should now be a menu bar on the Windows 10 style control panel
8. Right Click on the menu bar
9. Select the "StExBar" extension

If you de-select "Always Show Menus" from the view options, StExBar will be disabled again and you'll have to repeat the process.

So... got it working, but its a bit clunky...

**Comment by stefankueng** [comment]
unfortunately I think this will get even worse with every new Windows version - I think MS wants to get rid of DeskBands completely.

**Comment by gerhard4** [comment]
That Control Panel trick as described above by Stefan did work for me. But now everything in Explorer is switched to the Windows 10 UI, specifically the ribbon and the context menu. 

For now that's ok for me, but do you know a way back to the Windows 11 Explorer UI? I tried to uncheck the StExBar in the same location, but that just hides StExBar, it doesn't go back to the Windows 11 Explorer UI. There's something else that seems to have changed permanently.

**Comment by stefankueng** [comment]
I'm on Win11 beta channel. The explorer there has tabs. And there the UI is a little bit better, it keeps the Win11 UI style. But dark mode has some problems: right above the StExBar deskband a menu is shown which is always in bright mode, so it looks ugly.

**Comment by gerhard4** [comment]
After a bit I found out that the Explorer window where I enabled StExBar stays in Win10 mode, but any new Explorer window opens in Win11 mode (including the context menu), with the StEx toolbar. And if I disable the StExBar (again using the Control Panel trick), any new window looks like before installing StExBar (normal Win11 UI).

So with this small Control Panel work-around, it all works as it should. (I don't use dark mode.)

**Comment by jimtut** [comment]
@stefankueng , do you think a script like shown here could work, without the Control Panel steps?
https://docs.microsoft.com/en-us/windows/win32/shell/shell-showbrowserbar

I think I found a CLSID of 6c7a85a7-27c6-49ce-98b2-a8479b0dd63d for StExBar, and I tried it, but got the error "Class doesn't support Automation".  Is that your CLSID? Is "automation" something simple like a flag you can set when building that would allow your app to accept Shell commands like that?

**Comment by stefankueng** [comment]
@jimtut yes, that's the CLSID of the stexbar deskband object. But as you've discovered, you can't just use that.
Also the API you linked to is not usable in this context.

To get the StExBar to show in explorer the easy way:

1. open explorer
2. select the "control panel" item in the tree view
3. -> explorer reverts back to the 'old' view
4. now you can use the menu in explorer to enable the StExBar, and it will show up in the new explorer views as well

**Comment by jimtut** [comment]
Yes, I had seen that workaround above and used that successfully on Windows 11.  I was just thinking if there was a way to use an API to enable that "workband", which I thought might be the same as the bar in that API.  The API says:

> The object must be registered as an Explorer Bar object with a CATID_InfoBand component category. For further information, see [Creating Custom Explorer Bars, Tool Bands, and Desk Bands](https://docs.microsoft.com/en-us/windows/win32/shell/band-objects).

Sounds a lot like what StExBar is, so it's too bad if it's actually a different class/object.

**Comment by ramy2345** [comment]
I just started using Windows 11; version 22H2. I wanted to use StExBar, but the workarounds described here to enable the explorer bar didn't work. There seems to be no longer any option to enable the menu permanently nor to enable an explorer tools bar. Did I overlook anything? Does anyone has an idea hoe to use StExBar also in Win 11?

**Comment by jimtut** [comment]
I'm still on W11 21H2, but the "old" Control Panel is there, and you can enable toolbars in the View menu there.  Can you include a pic of your Control Panel and its View menu?  Maybe it's been removed, maybe you're just missing it.

**Comment by ramy2345** [comment]
> I'm still on W11 21H2, but the "old" Control Panel is there, and you can enable toolbars in the View menu there. Can you include a pic of your Control Panel and its View menu? Maybe it's been removed, maybe you're just missing it.

Sure, attached some screenshots from selecting "Control Panel", then the "Control Panel" View itself, and finally the menu. Note that the only way I've found to activate the menu was to press and release the Alt-key. After that the menu was visible on the upper left. But nowhere I could see any option to activate any tool bar or similar.

![Scr_20230326_194307_Home](https://user-images.githubusercontent.com/128686511/227794454-079ac24c-6cd2-4c92-a1b1-7386cbf5ff45.png)
![Scr_20230326_194613_Control Panel](https://user-images.githubusercontent.com/128686511/227794463-94479ceb-20bc-4c7d-8fd9-5a3ee411532d.png)
![Scr_20230326_194642_Control Panel](https://user-images.githubusercontent.com/128686511/227794465-c5479c2f-44b5-4964-adba-1fed52af2d70.png)

**Comment by jimtut** [comment]
My menus are in the horizontal (left-to-right) structure shown in my pic below, and I definitely have the Toolbars menu, and StExBar available there.  So, either your version of Windows and the Control Panel has removed this feature, or maybe StExBar isn't actually installed?  Included a 2nd pic of the Control Panel "Programs and Features" tool that shows StExBar installed on my machine, for you to compare if it's really installed for you.

If this doesn't help, the future looks bleak for all of us users...

![image](https://user-images.githubusercontent.com/5229799/227799958-cf2fd947-d757-4906-875c-f5e9294436d6.png)

<img width="548" alt="image" src="https://user-images.githubusercontent.com/5229799/227800012-d18f246a-fd55-4c52-8011-c8e25b0229d8.png">

**Comment by francoross** [comment]
**Alt + V** and then you select StExBar from Toolbars

**Comment by RobertKlohr** [comment]
Here is another workaround if you want to revert to the older right-click menu options.

To revert to the old UI run the following command:
```shell
reg add "HKCU\Software\Classes\CLSID\{86ca1aa0-34aa-4e8b-a509-50c905bae2a2}\InprocServer32" /f /ve
``` 
To change it back run the following command:
```shell
reg add "HKCU\Software\Classes\CLSID\{86ca1aa0-34aa-4e8b-a509-50c905bae2a2}\InprocServer32" /f /ve
```

**Comment by jimtut** [comment]
Looks like this may have changed again in Windows Feature Experience Pack 1000.22700.1034.0, which seems to have come out in the last month.

For me, I had to follow these actions:

1. Start -> Control Pannel
2. Press the Up arrow in the toolbar (not your keyboard) twice to get a special Desktop view.
3. This enabled the View->Options menu, from which you can enable StExBar:

![image](https://github.com/user-attachments/assets/c5f23645-804b-45e0-be20-3c566247f636)

**Comment by ibay770** [comment]
I've had success with ExplorerPatcher or Start11, to bring back the win10 Ribbon, once you're done enabling it, you can switch back and it'll stay.

**Comment by michelverhagen** [comment]
I think it has stopped working completely now. I'm fully updated, Microsoft Windows 11 Enterprise, Version 10.0.22621 Build 22621. None of the above tricks work anymore. StExBar just isn't shown in the View -> Options menu (it just shows "Change folder and search options").

I really hope Stefan can somehow modify StExBar so it works on the latest Windows 11 because I really, really miss it!

**Comment by ibay770** [comment]
You can get it working, but its a process, download Startallback, right click the windows flag, and click settings. Then click explorer and choose the win10 ribbon option. You should have the ribbon back in explorer.  Enable stexbar and then you can delete startallback and itll still be in explorer. 

https://www.startallback.com/

**Comment by michelverhagen** [comment]
No, the same can be done with ExplorerPatcher (which I'm using). That also can set the ribbon back (and so can the reg patches above). The problem is that in the latest, fully updated Windows 11, StExBar simply doesn't show in the View -> Options menu.. It's just not there, and nothing in this thread makes it appear, so I think MS has finally somehow completely shut this functionality down.

**Comment by ibay770** [comment]
> No, the same can be done with ExplorerPatcher (which I'm using). That also can set the ribbon back (and so can the reg patches above). The problem is that in the latest, fully updated Windows 11, StExBar simply doesn't show in the View -> Options menu.. It's just not there, and nothing in this thread makes it appear, so I think MS has finally somehow completely shut this functionality down.

Even with Startisback? Haven't tried the latest win11, so I can't confirm.

**Comment by michelverhagen** [comment]
I'm a dumb bastard. I installed the x86 version of StExbar on Win11 64 bit doh! Installed 64 bit version and of course it all works fine again...

**Comment by VulcanTourist** [comment]
The earlier advice in this discussion from years ago, to select Control Panel from the left navigation pane and use that as a back door to the pre-11 menu and options is no longer valid now in 2026.  Control Panel can still be found and selected, but the window that is displayed has only a title bar and no menu!

<img width="1049" height="488" alt="Image" src="https://github.com/user-attachments/assets/941f4410-6ea4-4adf-851b-542df41aed39" />

I'd like my Explorer toolbars back....

**Comment by condition-oakland** [comment]
I also just lost the StExBar in windows 11, and am looking for a way to get it back, as none of the earlier advice seems to be working for me. 

I was able to enable the View menu as per jimtut's advice on Sep 20, 2024, but there is no down-arrow on the options button.

<img width="1135" height="160" alt="Image" src="https://github.com/user-attachments/assets/02d6fe1e-0208-4f9a-a423-5bc9c5218b6f" />
